storage/block/types.go

```diff
@@ -108,6 +108,7 @@ type FetchBlockMetadataResults interface {
 // FetchBlocksMetadataResult captures the fetch results for multiple blocks.
 type FetchBlocksMetadataResult struct {
 	ID     ident.ID
+	Tags   ident.TagIterator
```

Comment:
might be better to have this be `ident.Tags` too - the series has the slice anyway. This way we can avoid the `TagSliceIterator` alloc per ID, have only a single instance In `service` and re-use using `Reset()` method you added

Reply:
As discussed, unfortunately can't as the on disk representation needs to be returned with the same type and its much more efficient to return TagDecoder which won't allocate a []ident.Tag array.  They need to share the same type as the in series one.

Reply:
Sounds good.